pkg/crypto/certificatemanager/certificate_manager.go

```diff
@@ -159,3 +159,29 @@ func (m *Manager) GetTLSContext(ctx context.Context, tlsCtx *api.TLSContext) (ca
 	}
 	return "", "", "", err
 }
+
+// GetSecretString returns a secret string stored in a k8s secret
```

Comment:
This could have been split into a different PR for easier review.

Reply:
Maybe review 3 first commits separately, one-by-one?